src/algorithms/standard/peakdetection.cpp

```diff
@@ -165,6 +166,43 @@ void PeakDetection::compute() {
     }
   }
 
+  // remove peaks that are closer than 'minPeakDistance'
+  if (_minPeakDistance > 0 && peaks.size() > 1) {
+    
+    std::vector<int> deletedPeaks;
+    deletedPeaks.reserve(peaks.size());
+    Real minPos;
+    Real maxPos;
+    
+    // iterate following an amplitide hierarchy
+    std::sort(peaks.begin(), peaks.end(),
+          ComparePeakMagnitude<std::greater<Real>, std::less<Real> >());
+
+    int k = 0;
+    while (k < peaks.size() - 1) {
+      minPos = peaks[k].position - _minPeakDistance;
+      maxPos = peaks[k].position + _minPeakDistance;
+
+      for (int l = k+1; l < peaks.size(); l++) {
+        if (peaks[l].position > minPos && peaks[l].position < maxPos) 
+          deletedPeaks.push_back(l);
+      }
+
+      // delete peaks starting from the end so the indexes are not altered
+      std::sort(deletedPeaks.begin(), deletedPeaks.end(), std::greater<int>());
+      
+      for (int l = 0; l < deletedPeaks.size(); l++)
+        peaks.erase(peaks.begin() + deletedPeaks[l]);
+        
+      deletedPeaks.clear();
+      deletedPeaks.reserve(peaks.size());
+      k++;
+    }
+
+    std::sort(peaks.begin(), peaks.end(),
```

Comment:
In the case of ```_orderBy == "amplitude"``` and ```_minPeakDistance > 0``` the peaks are first sorted by magnitude, then sorted by position and then sorted by magnitude again. This can be optimized.